### ISSUE TITLE
Not supported by Mobile Iron in Android

### ISSUE THREAD

**Issue by alexrainman** [paulcbetts/ModernHttpClient#176]
Hi there,

Is it possible to use a different message handler in Android?

One of these will be ok:

java.net.HttpURLConnection
java.net.ssl.HttpsURLConnection
Android HttpClient
DefaultHTTPClient, using the standard Apache HttpClient library with the org.apache.http.package name.

Cheers!

**Comment by Cheesebaron** [comment]
Ehm? Why?

**Comment by alexrainman** [comment]
They say OkHttp is not standard. We are making a ticket requesting it so, we keep using Xamarin best practices to share http calls related code.

**Comment by csunwold** [comment]
Who is they? Did someone from Xamarin specifically tell you not to use
OKHttp or are you going strictly off documentation.

Using ModernHttpClient as is is almost a de facto standard given the wide
number of Xamarin apps using it.

**Comment by alexrainman** [comment]
When i said "they" i was talking about Mobile Iron. They dont support OkHttp on their wrapping technology for Android.

**Comment by alexrainman** [comment]
Guys, any idea on how to get message handler for one of those clients from PCL? I mean, i can implement an interface to implement in android and return me the handler ...

**Comment by paulcbetts** [comment]
No, this is not possible. Either use the built-in Xamarin HttpClient or uh stop using this terrible wrapper or giving this company money

**Comment by Coruscate5** [comment]
this issue can be closed - MobileIron now supports OkHttpClient (& ModernHttpClient). They don't however support certain abstractions on top of it (React native being one of them). 

I'm in unsupported territory, so whether or not my implementation actually works would be irrelevant for this issue

**Comment by alexrainman** [comment]
Yes, we wroked closely with MobileIron to provide support for OkHttp.

**Comment by juvenull** [comment]
alexrainman, I have a similar problem. what the solution?

**Comment by alexrainman** [comment]
Well right now its compatible. We worked with MobileIron to make OkHttp compatible with their tunneling technology so you should be ready to go.
